SDK/simVis/Headless.h

```diff
@@ -43,29 +43,12 @@ namespace simVis
 
 #else
 
-#include <X11/Xlib.h>
-
 namespace simVis
 {
   /** Returns true when executing in a headless display environment */
-  inline bool isHeadless()
```

Comment:
We had a very good reason at one time why we wanted isHeadless() to be header-only, for external unit tests.  I'm not sure if that reason still exists, I'll need to test on our side on Windows and Linux.

Reply:
I wasn't sure if that was the case, and didn't see any particular in the SDK itself.  So I went with the least 'polluting' approach for the X11 on Linux.  But it could also be solved via adding more `#undefine` to the bottom of the file if it still needs to be a header only file

Reply:
Depending on include order, the `#undef` could have bad consequences too.

Reply:
I agree, hence my move to try to hide all the defines in the compilation unit.  Having X11 brought in also tends to throw 'odd' errors.  It took me a few moments to determine why `typedef std::complex<double> Complex;` was carping about typedefing an integer literal.  

It may be worthwhile to remove Headless.h from install all together.  It's more test code than production, as my client code would be relying on either the pipeline or the frameworks I'm using to determine which unit tests I'm running

Reply:
Agreed, those legacy headers and defines are a pain.

I'm not wholly opposed to removing it from the install, but if we were to deprecate simVis.h and remove on next pass, would that not make it a non-issue and could keep it in the install?

Reply:
The issue of Headless.h including X11 would still exist on Linux systems.  The issue being X11 collides with other symbols in the SDK.  It makes it less likely for it to occur, but I believe this is a minimum set of the problem:

```
#include <simVis/Headless.h>  //#define Complex 0
#include <simVis/Platform.h>  //typedef std::complex<double> Complex

int main() {
  return 0;
}
```

`Platform.h` includes `simCore/EM/RadarCrossSection.h` which includes `simCore/Calc/Random.h` which will cause the X11 #defines to come in.

Reply:
The primary concern on our side with making Headless.h into a .h/.cpp (not header-only) on our side is dependencies.  Linking against libsimVis brings in some heft dependencies that alter the redistributability of the target object.  It's not a big deal if you're already using simVis, but we have several apps that use this that don't rely on simVis.

Perhaps another solution is to move it to simCore eventually.  We could also bite the bullet and either link in simVis, or create a copy of `isHeadless()`.

Is your primary concern the `Complex` define?  I'd have no issues adding a `#undef Complex`, but there are other nasty defines from the X11 headers besides that one that can cause issues.

Reply:
What about simUtil?  simCore it seems semi out of place, but I can understand the desire to not want a large set of external dependencies for a single method, so simUtil may have the same issue for you.

The current example is `Complex` but that is really just the exposure of the underlying issue of isHeadless() pulling in a header that has a large amount of defines that then pollute the rest of the SDK causing compilation issues depending on include order.  This then becomes a maintenance issue as every symbol has to be checked against the X11 libs to see if there is a collision.  As a client of the SDK, I would expect to be able to include any arbitrary set of SDK headers, in arbitrary order, in my software without a break in compile.  

The removal of the wrap-up headers will do a job of reducing this as an issue, the only reason I saw it was my use of the `simVis.h`, and I eventually did remove that header to go back to just the required headers themselves.  So the fix could even just be putting 'here be dragons' right before the X11 include :D